src/awkward/contents/recordarray.py

```diff
@@ -362,42 +366,22 @@ def _getitem_range(self, where):
                 backend=self._backend,
             )
 
-    def _getitem_field(self, where, only_fields=()):
-        if len(only_fields) == 0:
-            return self.content(where)
+    def _getitem_field(self, where: str | SupportsIndex) -> Content:
+        return self.content(where)
 
-        else:
-            nexthead, nexttail = ak._slicing.headtail(only_fields)
-            if isinstance(nexthead, str):
-                return self.content(where)._getitem_field(nexthead, nexttail)
-            else:
-                return self.content(where)._getitem_fields(nexthead, nexttail)
-
-    def _getitem_fields(self, where, only_fields=()):
+    def _getitem_fields(self, where: list[str | SupportsIndex]) -> Content:
         indexes = [self.field_to_index(field) for field in where]
         if self._fields is None:
             fields = None
         else:
             fields = [self._fields[i] for i in indexes]
 
-        if len(only_fields) == 0:
-            contents = [self.content(i) for i in indexes]
-        else:
-            nexthead, nexttail = ak._slicing.headtail(only_fields)
-            if isinstance(nexthead, str):
-                contents = [
-                    self.content(i)._getitem_field(nexthead, nexttail) for i in indexes
-                ]
-            else:
-                contents = [
-                    self.content(i)._getitem_fields(nexthead, nexttail) for i in indexes
-                ]
-
+        contents = [self.content(i) for i in indexes]
         return RecordArray(
             contents, fields, self._length, parameters=None, backend=self._backend
         )
 
```

Comment:
I removed these because it was easy to, and functionally the behaviour should match. However, I am assuming that this would be a regression. Though I can't see any mention of it in v1, I am somewhat confident that this is likely an optimisation to avoid slicing each field of an array if the fields are ultimately dropped? If so, I'd likely want to revert this change.

Reply:
The `only_fields` argument was not in v1; it was added in v2 as a way to avoid code duplication. It's relevant for selecting a field or fields from a call stack that goes through `_getitem_next`, rather than a direct field-selection. (That is, the `only_fields` aregument is relevant for `__getitem__` calls in which a tuple item is a string, possibly after some non-string tuple items.)

That code path hasn't gone away—it can't be dead code—I'd strongly expect some tests to fail if you just remove `only_fields`.

But for all that, I can't remember exactly what it _means_. The name, "only fields" is not ringing a bell. I'd have to examine it more closely to remind myself.

Reply:
The code path is still in use. From my understanding, it's an optimisation that means we lookup additional field slices in order to avoid intermediate slices that need to copy data (mostly indices, I think).

The optimisation is, I think, here: 9ae35faf. 

https://github.com/scikit-hep/awkward/blob/db76e5fcf38560afd4f134908cc32d3ee1811e89/src/awkward/contents/content.py#L308-L318

It rewrites the slice tuple to perform all field slices early. Maybe we should do that anyway as part of our normalisation.

The tests pass (meaning that my understanding is hopefully not incorrect) if one removes the mechanism and instead does something like

```python
nexthead, nexttail = split_head_tail(tail)
record._getitem_fields(head)
```

I'll look into this.